LICENSE-APACHE-2.0

```diff
@@ -0,0 +1,201 @@
+      any Contribution intentionally submitted for inclusion in the Work
+      by You to the Licensor shall be under the terms and conditions of
+      this License, without any additional terms or conditions.
+      Notwithstanding the above, nothing herein shall supersede or modify
+      the terms of any separate license agreement you may have executed
+      with Licensor regarding such Contributions.
+
+   6. Trademarks. This License does not grant permission to use the trade
+      names, trademarks, service marks, or product names of the Licensor,
+      except as required for reasonable and customary use in describing the
+      origin of the Work and reproducing the content of the NOTICE file.
+
+   7. Disclaimer of Warranty. Unless required by applicable law or
+      agreed to in writing, Licensor provides the Work (and each
+      Contributor provides its Contributions) on an "AS IS" BASIS,
+      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+      implied, including, without limitation, any warranties or conditions
+      of TITLE, NON-INFRINGEMENT, MERCHANTABILITY, or FITNESS FOR A
+      PARTICULAR PURPOSE. You are solely responsible for determining the
+      appropriateness of using or redistributing the Work and assume any
+      risks associated with Your exercise of permissions under this License.
+
+   8. Limitation of Liability. In no event and under no legal theory,
+      whether in tort (including negligence), contract, or otherwise,
+      unless required by applicable law (such as deliberate and grossly
+      negligent acts) or agreed to in writing, shall any Contributor be
+      liable to You for damages, including any direct, indirect, special,
+      incidental, or consequential damages of any character arising as a
+      result of this License or out of the use or inability to use the
+      Work (including but not limited to damages for loss of goodwill,
+      work stoppage, computer failure or malfunction, or any and all
+      other commercial damages or losses), even if such Contributor
+      has been advised of the possibility of such damages.
+
+   9. Accepting Warranty or Additional Liability. While redistributing
+      the Work or Derivative Works thereof, You may choose to offer,
+      and charge a fee for, acceptance of support, warranty, indemnity,
+      or other liability obligations and/or rights consistent with this
+      License. However, in accepting such obligations, You may act only
+      on Your own behalf and on Your sole responsibility, not on behalf
+      of any other Contributor, and only if You agree to indemnify,
+      defend, and hold each Contributor harmless for any liability
+      incurred by, or claims asserted against, such Contributor by reason
+      of your accepting any such warranty or additional liability.
+
+   END OF TERMS AND CONDITIONS
+
+   APPENDIX: How to apply the Apache License to your work.
```

Comment:
Whoops, hit the wrong button.

As I understood it, those are just non-legally binding appendixes. I checked into the licenses distributed by the Rust project with `rustc` and they seem to have fully deleted this appendix

Reply:
Because this appendix seems to talk about adding this notice to *each and every file* which seems a little excessive?  
The SPDX identifier is added to each and every Cargo.toml now and has been symlinked into the directory. There is no way you could confuse the license when looking at the full work IMO.

When looking at other projects following the same dual-licensing scheme (I'm again looking at `rustc` itself), even components such as `miri` don't seem to have the header in every file. 

They have the licenses next to the Cargo.toml and the SPDX identifier in the Cargo.toml. And that's where they seem to stop

Reply:
Ah, I only intended to mean adding a copyright notice to _somewhere_ in the package(s) (maybe most likely in a `NOTICE` file, which is mentioned in 4.(d) of the License), rather than in every source file, but yeah the appendix can most naturally be read as the latter, which I agree is excessive.

Reply:
Can you show me an example of how you'd expect that notice file to look like?

I can't see one in the `futures-rs` repository

Reply:
Hmm, no projects under the rust-lang GitHub organisation seem to have the `NOTICE` file, actually. I myself have had one in a personal project and I think I've modeled it after certain projects, but I cannot remember one.

Anyway, for compatibility with the Rust ecosystem, you might instead want to follow the precedents of major Rust projects.

rust-lang/rust used to have a copyright notice in a `COPYRIGHT` file and have later changed the file to refer to the version control history or <https://thanks.rust-lang.org> (<https://github.com/rust-lang/rust/commit/f1de27fe6b73a8463fb49d4d2c88a24fa7031b82>, <https://github.com/rust-lang/rust/commit/c57deb923e56e04634b7e4b3fdc0f84ea0d41ed4>).

But rust-lang/rust is a big project with a lot of vendored subprojects and its convention might not necessarily suit to Kitsune.

On the other hand, rust-lang/futures-rs… OMG they've directly edited the [`LICENSE-APACHE`][futures-rs_LICENSE-APACHE] file and replaced the placeholder in the appendix with an actual copyright notice!

Well, so it turned out that fewer projects in the ecosystem have separate copyright notice for the Apache License than I expected, and none has the `NOTICE` file AFAICT. I think the notice isn't mandatory and that maybe you can skip it if you don't care about credits in Derivative Works.


[futures-rs_LICENSE-APACHE]: <https://github.com/rust-lang/futures-rs/blob/0fa6a1af9089e587a8fde1882284d08436b0c05c/LICENSE-APACHE#L189-L190>